common/djangoapps/entitlements/api/v1/views.py

```diff
@@ -274,30 +274,43 @@ def destroy(self, request, uuid):
             )
 
         if is_refund and entitlement.is_entitlement_refundable():
-            with transaction.atomic():
-                # Revoke and refund the entitlement
-                if entitlement.enrollment_course_run is not None:
-                    self._unenroll_entitlement(
-                        entitlement=entitlement,
-                        course_run_key=entitlement.enrollment_course_run.course_id,
-                        user=request.user
+            # Revoke the Course Entitlement and issue Refund
+            log.info(
+                'Entitlement Refund requested for Course Entitlement[%s]',
+                str(entitlement.uuid)
+            )
+            refund_status = refund_entitlement(course_entitlement=entitlement)
+
+            if refund_status:
+                with transaction.atomic():
+                    entitlement.expired_at_datetime = timezone.now()
+                    entitlement.save()
+
+                    log.info(
+                        'Set expired_at to [%s] for course entitlement [%s]',
+                        entitlement.expired_at,
+                        entitlement.uuid
                     )
```

Comment:
Nit: If the motivation here is to be able to debug problems with entitlement refunds, we might want to add something to this message specifying that expired_at is being set due to refund request.

Reply:
This is covered by the other log statements and can be gleaned in the logs from the sequence of messages.